### ISSUE TITLE
ResolverOpts::attempts is actually number of re-try attempts

### ISSUE THREAD

**Issue by lukaspustina** [bluejekyll/trust-dns#1176]
**Describe the bug**
I'm not sure if this is really a bug, but IMHO the documentation of `ResolverOpts` suggests a different meaning of the field `attempts`:
* Documentation says: 
  > attempts: usize - Number of attempts before giving up. Defaults to 2
* The code seems to indicate otherwise though. The number of attempts is decremented _after_ the first try; so the field represents the number of re-try attempts -- `retry_dns_handle.rs:74`
  ```
    fn poll(mut self: Pin<&mut Self>, cx: &mut Context) -> Poll<Self::Output> {
        // loop over the future, on errors, spawn a new future
        //  on ready and not ready return.
        loop {
            match self.future.poll_unpin(cx) {
                Poll::Ready(Err(e)) => {
                    if self.remaining_attempts == 0 {
                        return Poll::Ready(Err(e));
                    }

                    self.remaining_attempts -= 1;
                    // TODO: if the "sent" Message is part of the error result,
                    //  then we can just reuse it... and no clone necessary
                    let request = self.request.clone();
                    self.future = self.handle.send(request);
                }
                poll => return poll,
            }
        }
  ```

**To Reproduce**
Count the amount of UDP packets sent with Wireshark or tcpdump. But it's actually in the code.

**Expected behavior**
From the documentation I expected to see up to `ResolverOpts::attempts` packets. 

**Version:**
Crate: resolver
Version: 0.19.5

**Comment by bluejekyll** [comment]
Interesting, seems like a poor interpretation of what that value means! Yes, I would expect that is being more used as a retry counter rather than more strictly as the total number of attempts, including the first request.

I suppose we can either fix the code to use this and include the first attempts as well?

**Comment by lukaspustina** [comment]
I don't mind either fix. I guess, it's easier and friendlier to current users of the crate to just clarify the documentation since a re-interpretation in the code could possibly brake existing usage of the parameter.

**Comment by bluejekyll** [comment]
agreed. If you'd like to submit a PR with the documentation updated, that would be really helpful!

**Comment by lukaspustina** [comment]
@bluejekyll: Done.

**Comment by bluejekyll** [comment]
Fixed in #1180
